### ISSUE TITLE
Does not work with non-ASCII characters

### ISSUE THREAD

**Issue by bortzmeyer** [martianhi/mastodon.el#32]
In mastodon-mode, non-ASCII characters in usernames or in toots are displayed by escape sequences of the UTF-8 (\303\251 instead of é).

mastodon.el just git pulled

GNU Emacs 24.5.1 (x86_64-pc-linux-gnu, GTK+ Version 3.18.9) of 2016-04-17 on lgw01-04, modified by Debian (can display Unicode just fine)

**Comment by tbje** [comment]
I have the same problem. The buffer is UTF-8 (there is a U on the mode-line that gives the information) so I assume it's something going wrong earlier in the process?

**Comment by alexjgriffith** [comment]
I am not sure if this issue has been fixed yet.

The strings are parsed and encoded as utf-8. To represent the string visually in emacs, you can use `decode-coding-string` or `decode-coding-region`.

Most standard symbols are already included in emacs:
``` elisp
(insert (decode-coding-string "\n;; \303\251ve" 'utf-8)) 
;; éve
```
If you have emojify installed it will also work with emojis:
``` elisp
(emojify-mode 't)
(insert (decode-coding-string "\n;; sample emoji \360\237\230\211" 'utf-8))
;; sample emoji 😉
```

**Comment by jdenen** [comment]
It has not been fixed yet and if this solves the problem, I will find a way to buy you a beer. This particular issue has been a giant PITA.

**Comment by alexjgriffith** [comment]
I've implemented a fix in my version. Will push once i've cleaned it up a bit
![screenshot 2017-04-22 14 49 58](https://cloud.githubusercontent.com/assets/6475397/25307221/13c09f66-276b-11e7-9a40-f7b445d9b4f6.png)

**Comment by jdenen** [comment]
Beautiful.

**Comment by jdenen** [comment]
Fixed by #53

**Comment by bortzmeyer** [comment]
Works beautifully, many thanks.
